journal/src/main/java/io/camunda/zeebe/journal/file/SegmentsManager.java

```diff
@@ -299,6 +308,15 @@ private void prepareNextSegment() {
     nextSegment = CompletableFuture.supplyAsync(() -> createUninitializedSegment(descriptor));
   }
 
+  Collection<Segment> getSegments(final long index) {
```

Comment:
:wrench: May be rename to `getTailSegments` and/or add a one line doc.

Reply:
:+1: Good catch, this is not descriptive at all :sweat_smile: I think tail is too tied to the `NavigableMap` concept, but then again something like `getSegmentsWithIndexGreaterThanOrEqualTo` is a mouthful :smile: